dashboard/app/controllers/api_controller.rb

```diff
@@ -370,11 +372,13 @@ def teacher_panel_progress
       student_progress = section.students.order(:name).map do |student|
         ScriptLevel.summarize_as_bonus_for_teacher_panel(lesson.script, bonus_level_ids, student)
       end
+
+      teacher_progress = ScriptLevel.summarize_as_bonus_for_teacher_panel(lesson.script, bonus_level_ids, current_user)
     else
       return head :bad_request
     end
 
-    render json: student_progress
+    render json: student_progress.push(teacher_progress)
```

Comment:
(very minor) I think having the teacher be the first element returned is more natural, but very minor, up to you.